config/registry/aws/modules/lambda-function.md

```diff
@@ -0,0 +1,110 @@
+---
+title: "aws-lambda"
+linkTitle: "aws-lambda"
+date: 2021-12-02
+draft: false
+weight: 1
+description: Creates a lambda deployment via opta.
+---
+
+*NOTE* The lambda module is currently in beta, with only the following functionality. Enhanced features and integration 
+with other opta modules will be added in subsequent releases (and expedited as per user need). Please 
+[reach out to the opta team](https://slack.opta.dev/) for any questions/requests.
+
+Creates an [AWS Lambda](https://docs.aws.amazon.com/lambda/index.html) deployment via opta from the given zip file plus
+much more setup. Currently, it also handles:
+
+* Setting up a cloudwatch log group for your function logs
+* Setting up an iam role for your lambda through which to grant permissions to.
+* Exposing the lambda to a public uri if you so wish
+* Allowing the user to pass in IAM policies to give to the lambda
+* Setting up security group and network location if you have the `aws-base` module in your environment.
+
+tl;dr -- this will deploy a hello world lambda
+```yaml
+name: testing-lambda
+org_name: myorg
+providers:
+  aws:
+    region: us-east-1
+    account_id: XXXXXXXXX
+modules:
+  - type: lambda-function
+    expose_via_domain: true
+```
+
+## Adding your Own Function
+Opta supports the [zip file form](https://docs.aws.amazon.com/lambda/latest/dg/configuration-function-zip.html) of
+lambda deployments. For this deployment, simply zip up your code into a zip file and pass its location in via the
```

Comment:
So I like the idea of a default deployment lambda but it shouldn't be baked into opta code. We should provide a link to an example zip file in our docs and tell the user "if you wanna try this out just download this zip and put a link to it in your opta.yml".
And then, we should make the `filename` field required.